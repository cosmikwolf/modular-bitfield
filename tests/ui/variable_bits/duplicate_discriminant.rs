use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[variable_bits(8, 16)]
enum TestData {
    #[discriminant = 0]
    Small(u8),
    #[discriminant = 0]  // Duplicate discriminant
    Medium(u16),
}

fn main() {}