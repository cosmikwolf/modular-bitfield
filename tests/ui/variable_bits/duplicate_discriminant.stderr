error: duplicate discriminant value 0 (variant Medium conflicts with previous variant)
 --> tests/ui/variable_bits/duplicate_discriminant.rs:8:5
  |
8 | /     #[discriminant = 0]  // Duplicate discriminant
9 | |     Medium(u16),
  | |_______________^
